### DISCUSSION TITLE
Error using binary UUID

### DISCUSSION THREAD

**Discussion by pimjansen** [api-platform/api-platform#2501] [Q&A]
Im not sure if it is kinda a bug or im just doing it wrong. Im using Ramsey UUID Doctrine extension to store binary uuids in my MySQL 8 database. This all works well and i can insert them and reading them is also fine using an ApiPlatform resource.

However it goes wrong when i try to do a GET on a resource with an invalid UUID. For example when i query on ID `1234` it gives me this error with a 500 response:
`Could not convert database value \"123\" to Doctrine Type uuid_binary`

I expect it to return me a 400 or maybe even a 404 because it does not know it.

```
#[ApiResource(
    operations: [
        new Get(),
        new GetCollection(),
    ],
    routePrefix: '/v1',
)]
#[ORM\Entity()]
#[ORM\Table(name: '`user`')]
class User
{
    #[ApiProperty(identifier: true)]
    #[ORM\Id, ORM\Column(name: 'user_id', type: 'uuid_binary')]
    #[ORM\GeneratedValue(strategy: 'CUSTOM')]
    #[ORM\CustomIdGenerator(class: UuidGenerator::class)]
    #[Assert\Uuid]
    public string $id;
```

Thanks!

**Comment by adnanjaw** [comment]
@pimjansen https://github.com/api-platform/core/issues/1309

**Reply by pimjansen** [reply]
This is an old solved topic however it seems still the case
